### ISSUE TITLE
Multi column headers break the Grid a11y cell announcement

### ISSUE THREAD

**Issue by mbechev** [telerik/kendo-angular#4875]
### Description

Having [multiple column headers](https://www.telerik.com/kendo-angular-ui/components/grid/columns/headers#multi-column-headers) in the Grid (2 or more) breaks NVDA and JAWS' ability to read cell content correctly.

Specifically, it reads the cell value as expected, but doesn't announce its corresponding column.
The cells in the green area are announced correctly, but the ones in the red area are incorrect.

### Steps To Reproduce

1. Open this demo - [https://stackblitz.com/edit/angular-jfjpjgce](https://stackblitz.com/edit/angular-jfjpjgce)
2. Start NVDA or JAWS and check what is announced for cells

### Screenshots or video

Notice how the **Quantity** cell is announced. The mentioned '*Weight*' in the speech viewer is actually the next column outside the group.

[https://github.com/user-attachments/assets/8e719729-2a04-4989-9fad-eeea060369a1](https://github.com/user-attachments/assets/8e719729-2a04-4989-9fad-eeea060369a1)

### Actual Behavior

Having multiple column header groups breaks the cell a11y, confusing the columns by reading an incorrect one.

### Expected Behavior

The cells need to be announced with their correct column value.

Setting *role="columheader*" to the  elements (group column header) in the Grid reads the columns correctly:

[https://github.com/user-attachments/assets/9041ca97-e825-4292-9c35-5553ceaddd7f](https://github.com/user-attachments/assets/9041ca97-e825-4292-9c35-5553ceaddd7f)
